src/idpi/config.py

```diff
@@ -0,0 +1,64 @@
+"""global configuration for idpi."""
+
+# Standard library
+from typing import Any, Literal
+
+config: dict = {}
+
+
+class set:
+    """Temporarily set configuration values within a context manager.
+
+    Parameters
+    ----------
+    config : dict, optional
+        dictionary object use to hold the configuration.
+        Default will use the config object in this module
+    **kwargs :
+        the configuration key-value pairs to set.
+
+    """
+
+    def __init__(self, config: dict = config, **kwargs):
+        self._config = config
+        self._record: list[tuple[Literal["insert", "replace"], str, Any]] = []
+
+        if kwargs:
+            for key, value in kwargs.items():
+                if key in self._config:
+                    self._record.append(("replace", key, self._config[key]))
+                else:
+                    self._record.append(("insert", key, None))
+
+                self._config[key] = value
+
+    def __enter__(self) -> dict:
+        return self._config
```

Comment:
I don't see the use case for returning the updated config since the global/passed config is mutated in place. More importantly, the actual mutation should happen in the enter function and not in the init. Maybe this calls for the context manager to be implemented using the function approach?

Reply:
Returning a global config is an internal detail, and not part of the contract. No one is accessing directly the global object, the internal implementation could be creating clones of it. 
Why should the actual mutation happen in the init?

Reply:
> Why should the actual mutation happen in the init?

Because the context manager is implemented using a class, it's actually not forbidden to dissociate the instantiation from the entering of the context e.g.
```
mgr = set(foo="bar")
...
with mgr:
    pass
```
The mutation would be expected to happen in the with block. However, with the current implementation, it would happen as soon as the init function is called.

Reply:
I see, thanks for the clarification. I am changing this to use a contextmanager decorator